python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_asset_group.py

```diff
@@ -521,3 +524,121 @@ def follows_o2(o2):
     assert result.output_for_node("middle_asset", "o1") == "foo"
     assert result.output_for_node("follows_o2") == "foo"
     assert result.output_for_node("start_asset") == "foo"
+
+
+def test_assets_prefixed_single_asset():
+    @asset
+    def asset1():
+        ...
+
+    result = AssetGroup([asset1]).prefixed("my_prefix").assets
+    assert result[0].asset_key == AssetKey(["my_prefix", "asset1"])
+
+
+def test_assets_prefixed_internal_dep():
+    @asset
+    def asset1():
+        ...
+
+    @asset
+    def asset2(asset1):
+        del asset1
+
+    result = AssetGroup([asset1, asset2]).prefixed("my_prefix").assets
+    assert result[0].asset_key == AssetKey(["my_prefix", "asset1"])
+    assert result[1].asset_key == AssetKey(["my_prefix", "asset2"])
+    assert result[1].dependency_asset_keys == {AssetKey(["my_prefix", "asset1"])}
+
+
+def test_assets_prefixed_external_dep():
+    @asset
+    def asset1():
+        ...
+
+    upstream_group = AssetGroup([asset1]).prefixed("upstream_prefix")
+
+    @asset
+    def asset2(asset1):
+        del asset1
+
+    result = AssetGroup([asset2]).prefixed("my_prefix", upstream_groups=[upstream_group]).assets
+    assert len(result) == 1
+    assert result[0].asset_key == AssetKey(["my_prefix", "asset2"])
+    assert result[0].dependency_asset_keys == {AssetKey(["upstream_prefix", "asset1"])}
+
+
+def test_assets_prefixed_disambiguate():
+    @asset(name="apple")
+    def asset1():
+        ...
+
+    upstream_group = AssetGroup([asset1]).prefixed("core")
+
+    @asset(name="apple")
+    def asset2():
+        ...
+
+    @asset(ins={"apple": AssetIn(namespace="core")})
```

Comment:
We probably will need to change `namespace` -> `prefix` in a later PR.

Reply:
👍